### DISCUSSION TITLE
New named role: Issue Wrangler

### DISCUSSION THREAD

**Discussion by divya-mohan0209** [kubernetes/website#38861] [Ideas]
# Context

In light of the many issues that await triage and to reduce the onus on formal approvers/reviewers, it'd be helpful to have a dedicated person to wrangle issues every week.  This role would also help bridge the gap between an organizational contributor and a reviewer. 

## Role description, a.k.a. what does success look like in this role?

- Ensuring the created issue is worded & titled correctly to provide adequate information for contributors.
- Ensuring the created issue does not fall under the support category and assigning the "triage/accepted" label.
- Ensuring the issue is tagged with appropriate sig/area/kind labels.
- Monitoring for stale & rotten issues within the k/website repo.

**_Nice to have: Maintenance of the [issues board](https://github.com/orgs/kubernetes/projects/72/views/1)_** 

## Requirements, a.k.a, what does somebody need to do to become an issue wrangler?
- Kubernetes org member
- At least 15 valuable contributions to Kubernetes (of which, there needs to be a certain amount to k/website).
- Already doing the role in an informal capacity 

## Where does it fit on the contributor ladder? 
- In between a contributor and a reviewer.
- The next step for someone who assumes the role & demonstrates capability would be to shadow a PR Wrangler and start informally reviewing PRs.

## Implementation:
- Create a roster of existing folks who triage issues.
- Pilot a shadow program & gauge interest
- Pass on the mantel if there is interest.
- Repeat.

## Are any extra privileges required for this role?
- No. Kubernetes org members can add the triage/accepted label & any other labels as they see fit to issue.

**Comment by a-mccarthy** [comment]
@divya-mohan0209 I really like this idea! I think it would be very valuable to share the load of issue management as well as have folks take responsibility for doing it. 

@natalisucks had mentioned starting up an issue triage meetings as a way to help review/prioritize issues more regularly. How do you see the Issue Wrangler and triage meeting working together?

**Comment by divya-mohan0209** [comment]
So, my vision for this role was to eventually lead those meetings by managing the triage board. Which is why I mentioned it as a nice to have. I don't know if that's entirely possible though.

**Comment by sftim** [comment]
I think we could have one _or more_ issue wranglers assigned. It's great to start with 1.

**Reply by divya-mohan0209** [reply]
I agree; however, if we overload one person, it might be a bit much.

**Comment by shaneutt** [comment]
Would we consider an alternative name for the role, like "Issue Officer" or "Issue Shepherd" perhaps? For some reason "wrangler" just sounds off to me :thinking:

**Reply by divya-mohan0209** [reply]
> Would we consider an alternative name for the role, like "Issue Officer" or "Issue Shepherd" perhaps? For some reason "wrangler" just sounds off to me 🤔

I don't have a particular preference, tbh. I'm good with changing the name of the role if everybody else is of the same opinion as well?

**Comment by natalisucks** [comment]
@shaneutt Issue Wrangler matches in terms of our [PR Wrangler role](https://github.com/kubernetes/website/wiki/PR-Wranglers), so I'd personally be up for keeping these the same as the community generally knows what our PR Wrangler schedule is and its purpose. With similar naming, we hope this new role becomes just as familiar within the community

**Comment by divya-mohan0209** [comment]
Added to the [discussion agenda](https://docs.google.com/document/d/1emuO4nmaQq3K8JZ9-MQeIygtrCPO9kWv7U7RzTaW4F8/edit?usp=sharing) for 05/16 meeting to seek suggestions & improvements from the wider community.
Emaiul & slack discussion links to follow.

**Comment by seokho-son** [comment]
I agree with this new role suggestion! 
One question as a localization contributor. If we have the issue wrangler role, should issue wranglers cover issues related with localization as well? Or, should we limit the coverage of issue wranglers only for non-localization issues and let each localization team handle localization issues?

**Reply by divya-mohan0209** [reply]
Thank you for your question, @seokho-son !

To keep the scope of work consistent across our wrangler roles, the idea is for the issue wrangler role to only work with the main website issues i.e. the English localization. The issues for every other localization will be handled by the respective localization approvers/reviewers. I hope that answers the question?

**Comment by sftim** [comment]
Is this discussion ongoing?

**Reply by aj11anuj** [reply]
Ya, atleast a little discussion about it happens in every SIG Docs meeting.

**Comment by aj11anuj** [comment]
It's been around 9 months since this discussion has started, and still this topic is in discussion. I think we should not think that much about it and just implement it on small scale to see how it results and then do further discussion on it. 

I had started doing this Issue Wrangler work informally since past some weeks to the best of my knowledge and I'm learning on the go. I have developed a habit of checking this repo's issues and PR's almost every hour from morning to midnight and help out with labelling and discussion as much as i can. I'm also an active regular contributor of K8s/website repo. I am willing to be the first Issue Wrangler or Issue Wrangler Shadow, if i was selected.

**Comment by divya-mohan0209** [comment]
This is being worked on currently under #42273. We can close this discussion since the issue is also being tracked under [41601](https://github.com/kubernetes/website/issues/41601).
